### PR TITLE
docs: update installation guide

### PR DESCRIPTION
**Pull request by nitishfy**
### Description

Generally, when you're trying to install the apisix from the [installation guide](https://apisix.apache.org/docs/apisix/installation-guide/), sometimes you come across the `Address Bind already in use` error logs when trying to start the apisix server. This is likely caused due to the unavailability of the port. Mostly, it happens if you've installed the apisix using docker due to which the ports get equipped. Hence, when you try installing the apisix locally, you get the error logs. 

This PR mentions this minor information with solution to this problem. There is a little refactoring of the docs as well. Since, it is required to install the `etcd` first, the installation part of etcd is moved first, followed by installation of `apisix` server installation.

Fixes # (issue)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: docs/en/latest/building-apisix.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ APISIX_VERSION='3.9.0'
 Then, you can run the following command to clone the APISIX source code from Github:
 
 ```shell
-git clone --depth 1 --branch ${APISIX_VERSION} https://github.com/apache/apisix.git apisix-${APISIX_VERSION}
+git clone --depth 1 --branch master https://github.com/apache/apisix.git apisix-${APISIX_VERSION}
```
[the remark this answers is not in the document]
Comment (shreemaan-abhishek) [reply docs/en/latest/building-apisix.md:57]: I think separate installation steps can be provided in the VOTE email and this change can be kept.

File: docs/en/latest/installation-guide.md [hunk 2]
```diff
@@ -271,9 +272,11 @@ You can configure your APISIX deployment in two ways:
    apisix start -c <path to config file>
    ```
 
-APISIX will use the configurations added in this configuration file and will fall back to the default configuration if anything is not configured.
+APISIX will use the configurations added in this configuration file and will fall back to the default configuration if anything is not configured. Generally, APISIX gets installed at `/usr/local/apisix/` directory, so your configuration file will be present at `/usr/local/apisix/conf/` path.
+
+In case you get the Port binding logs when trying to run the APISIX server using `apisix start` command, it is most likely that the ports are being used by certain processes running on your machine. Try deleting the process using the port or configure the default listening port to other available port on your local machine.
```
Comment (kayx23) [inline docs/en/latest/installation-guide.md:277]: What `Port binding logs`? Be specific of the error

Comment (kayx23) [reply docs/en/latest/installation-guide.md:277]: `deleting the process` --> `killing the process`

File: docs/en/latest/installation-guide.md [hunk 3]
```diff
@@ -271,9 +272,11 @@ You can configure your APISIX deployment in two ways:
...
 
-For example, to configure the default listening port to be `8000` without changing other configurations, your configuration file could look like this:
+In order to configure the default listening port to be `8000` without changing other configurations, your configuration file could look like this:
```
Comment (kayx23) [inline docs/en/latest/installation-guide.md:279]: Why the change here? `For example` + `could` sound like a pair providing a sample config, but now the instruction sounds definitive, which it should not be.

**Review by kayx23 — COMMENTED** [review]
Please fix for CI linting error

File: docs/en/latest/installation-guide.md (hunk 3 above)
Comment (nitishfy) [reply docs/en/latest/installation-guide.md:279]: If you read the sentence above it that mentions `configure the default listening port to other available port on your local machine.`... this makes sense. We're showing how do you perform the last step mentioned in the documentation rather than giving you an example which IMO is vague at this point.

File: docs/en/latest/installation-guide.md (hunk 2 above)
Comment (nitishfy) [reply docs/en/latest/installation-guide.md]: done!

**Comment by nitishfy** [comment]
The linter issues have been fixed now

File: docs/en/latest/building-apisix.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (kayx23) [reply docs/en/latest/building-apisix.md:57]: So what's the verdict here?  `APISIX_VERSION` is removed in this change but it is still set in the step above. @nitishfy 

image

File: docs/en/latest/installation-guide.md [hunk 4]
```diff
@@ -271,9 +271,13 @@ You can configure your APISIX deployment in two ways:
    apisix start -c <path to config file>
    ```
 
-APISIX will use the configurations added in this configuration file and will fall back to the default configuration if anything is not configured.
+APISIX will use the configurations added in this configuration file and will fall back to the default configuration if anything is not configured. Generally, APISIX gets installed at `/usr/local/apisix/` directory, so your configuration file will be present at `/usr/local/apisix/conf/` path.
+
+In case you get the Port binding logs, that says `Bind address already in use` when trying to run the APISIX server using `apisix start` command, it is most likely that the ports are being used by certain processes running on your machine. Try killing the process which is using the port. You can find the PID of the process that is using the process by running the `sudo netstat -tulpn` command and later using `kill <process id>` command to kill the process.
```
Comment (kayx23) [inline docs/en/latest/installation-guide.md:276]:
```suggestion
If you are unable to start APISIX and observe `Bind address already in use` in logs, it means the ports APISIX wants to use are already occupied by other running processes. Upon reviewing the use of these processes, you may decide to kill these processes or configure APISIX to use other non-conflicting ports.
```

1. Some suggestion to reduce verbosity but you can edit again. If this is accepted, the sentence below needs to be removed as well.
2. You can add `netstat` back but note that flags can be distro / os specific (e.g. on mac `netstat: n: unknown or uninstrumented protocol`). Also `sudo` is not needed. If the audience don't have the needed privilege, they can escalate it themselves.

File: docs/en/latest/installation-guide.md (hunk 3 above)
Comment (kayx23) [reply docs/en/latest/installation-guide.md:280]: then `could` --> `should`

```suggestion
In order to configure the default listening port to be `8000` without changing other configurations, your configuration file should look like this:
```

File: docs/en/latest/building-apisix.md [hunk 5]
```diff
@@ -45,16 +45,10 @@ To build and package APISIX for a specific platform, see [apisix-build-tools](ht
...
-
-```shell
-APISIX_VERSION='3.9.0'
-```
-
 Then, you can run the following command to clone the APISIX source code from Github:
 
 ```shell
-git clone --depth 1 --branch ${APISIX_VERSION} https://github.com/apache/apisix.git apisix-${APISIX_VERSION}
+git clone --depth 1 --branch master https://github.com/apache/apisix.git
```
Comment (kayx23) [inline docs/en/latest/building-apisix.md:51]: @shreemaan-abhishek are these the agreed upon changes?

image

I dont have additional comments otherwise

Comment (shreemaan-abhishek) [reply docs/en/latest/building-apisix.md:51]: yes.

File: docs/en/latest/installation-guide.md [hunk 6]
```diff
@@ -271,9 +271,13 @@ You can configure your APISIX deployment in two ways:
    apisix start -c <path to config file>
    ```
 
-APISIX will use the configurations added in this configuration file and will fall back to the default configuration if anything is not configured.
+APISIX will use the configurations added in this configuration file and will fall back to the default configuration if anything is not configured. Generally, APISIX gets installed at `/usr/local/apisix/` directory, so your configuration file will be present at `/usr/local/apisix/conf/` path.
```
Comment (shreemaan-abhishek) [inline docs/en/latest/installation-guide.md:274]:
```suggestion
APISIX will use the configurations added in this configuration file and will fall back to the default configuration if anything is not configured. Generally, APISIX gets installed at `/usr/local/apisix/` directory, so the configuration file will be present at `/usr/local/apisix/conf/`.
```

File: docs/en/latest/installation-guide.md [hunk 7]
```diff
@@ -271,9 +271,13 @@ You can configure your APISIX deployment in two ways:
...
+
+If you are unable to start APISIX and observe `Bind address already in use` in logs, it means the ports APISIX wants to use are already occupied by other running processes. Upon reviewing the use of these processes, you may decide to kill these processes or configure APISIX to use other non-conflicting ports.
```
Comment (shreemaan-abhishek) [inline docs/en/latest/installation-guide.md:276]: I think this is pretty basic stuff, we can remove this.

File: docs/en/latest/installation-guide.md [hunk 8]
```diff
@@ -271,9 +271,13 @@ You can configure your APISIX deployment in two ways:
...
+
+Other way is to configure the default listening port to other available port on your local machine.
```
Comment (shreemaan-abhishek) [inline docs/en/latest/installation-guide.md:278]: ditto

**Comment by kayx23** [comment]
Do we need this PR still? The most beneficial change here as it seems is to re-arrange etcd and APISIX installation order.

Let's re-align what needs to be done here and move this effort forward. Thanks.

**Comment by kayx23** [comment]
🏓

**Comment by kayx23** [comment]
The only improvement beneficial here is to adjust the order of APISIX and etcd installation. This can be addressed in a separate PR. Closing this PR as it is stagnant.
